client/app/src/components/addressbook/addressbook.controller.js

```diff
@@ -63,6 +63,55 @@
       }
     }
 
+    self.addressBookAddContactRecord = function(name, address, callback, isRename, suppressNotice) {
+      self.getContacts();
+      if (self.trim(name) == "") {
+        self.showToast('This Contact Name is not valid', name, true);
+        return;
+      }
+      if (self.trim(address) == "" || ! self.isAddress(address)) {
+        self.showToast('This Contact Address is not valid', address, true);
+        return;
+      }
+
+      var newContact = { name: name, address: address };
+      if (!isRename && self.addressExists(address)) {
+        self.showToast('A Contact with this Address already exists', address, true);
+        return;
+      }
+      if (self.contactExists(name)) {
+        self.showToast('A Contact with this Name already exists', name, true);
+        return;
+      }
```

Comment:
Would be nice to pull all this error handling out into a method. Then follow the error handling with a single return if possible.

Might consider adding some unit tests for the logic as well to make sure everything is being handled appropriately.